### ISSUE TITLE
compile error with pytorch1.0.0 nightly

### ISSUE THREAD

**Issue by johnlanbor** [tianzhi0549/FCOS#29]
anaconda3/envs/py35torch1/lib/python3.5/site-packages/torch/include/ATen/Dispatch.h:15:17: error: switch quantity not an integer
     switch (TYPE) {                                                          \
                 ^
Project/FCOS/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp:71:3: note: in expansion of macro ‘AT_DISPATCH_FLOATING_TYPES’
   AT_DISPATCH_FLOATING_TYPES(dets.type(), "nms", [&] {
   ^
anaconda3/envs/py35torch1/lib/python3.5/site-packages/torch/include/ATen/Dispatch.h:16:44: error: could not convert ‘Double’ from ‘c10::ScalarType’ to ‘<type error>’
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Double, double, __VA_ARGS__)      \
                                            ^
anaconda3/envs/py35torch1/lib/python3.5/site-packages/torch/include/ATen/Dispatch.h:8:8: note: in definition of macro ‘AT_PRIVATE_CASE_TYPE’
   case enum_type: {                                \
        ^
Project/FCOS/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp:71:3: note: in expansion of macro ‘AT_DISPATCH_FLOATING_TYPES’
   AT_DISPATCH_FLOATING_TYPES(dets.type(), "nms", [&] {
   ^
anaconda3/envs/py35torch1/lib/python3.5/site-packages/torch/include/ATen/Dispatch.h:17:44: error: could not convert ‘Float’ from ‘c10::ScalarType’ to ‘<type error>’
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Float, float, __VA_ARGS__)        \
                                            ^
anaconda3/envs/py35torch1/lib/python3.5/site-packages/torch/include/ATen/Dispatch.h:8:8: note: in definition of macro ‘AT_PRIVATE_CASE_TYPE’
   case enum_type: {                                \
        ^
Project/FCOS/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp:71:3: note: in expansion of macro ‘AT_DISPATCH_FLOATING_TYPES’
   AT_DISPATCH_FLOATING_TYPES(dets.type(), "nms", [&] {

but when i copy code to maskrcnn benchmark, this error never happend
so, I think there may some version problem

**Comment by tianzhi0549** [comment]
@johnlanbor If you use exactly the same environment as maskrcnn-benchmark to FCOS, can you run it normally? We did not change any codes in these c++ files, and therefore I think it should work well in the environment where maskrcnn-benchmark can work.

**Comment by johnlanbor** [comment]
@tianzhi0549 I compile maskrcnn benchmark and FCOS in the same environment, but FCOS cant compile  correctly while benchmark can .

when i copy FCOS code to benchmark, and train with this setting: fcos_R_50_FPN_1x.yaml
when test woth 65000 pth can got AP:36
but I got loss(nan) when 79800 iter, is that because the lr?  I use 4Gpus , so the lr should be half of 0.01?

**Comment by tianzhi0549** [comment]
@johnlanbor You don't need to change any configs because of 4 GPUs. The number of GPUs does not change the batch size.

The loss might become NAN occasionally. You can try to continue training from your last checkpoint.
